doc/service.md

```diff
@@ -0,0 +1,85 @@
+What is a service?
+==================
+
+A service for Cozy is defined by these criteria:
+
+1. It's some JavaScript code
+2. That is embedded in Cozy-Proxy
+3. As express/americano middlewares
+4. And exposes a REST API in JSON
+5. to offer functionalities to cozy applications.
+6. This API follows the [JSON api specification](http://jsonapi.org/)
+7. And [Cozy guidelines](https://cozy.github.io/cozy-guidelines/)
+8. With URL prefixed by `/services/:service-name`.
+9. It benefits from Cozy authentication and permissions
+10. And workflows (code review, i18n).
+
+
+Rationale
+---------
+
+### 1. It's some JavaScript code
+
+JS is the lowest common denominator for all the code written at Cozy.
+Technically, a service can be written in something else (CoffeeScript or
+TypeScript) but it has to be compiled simply to JavaScript.
+
+### 2. That is embedded in Cozy-Proxy
+
+I can think to several services than can be useful: files, notifications,
+applications management, sharing, settings, indexer/search, etc.
+Running one node processus for each service will take a lot of RAM.
+[Simple Cozy](https://github.com/cozy/simple-cozy) has shawn that it's
+possible to run a nodejs processus with code from several git repository to
+use less memory.
+
+I think to embed the services in cozy-proxy and not cozy-data-system beacuse
+cozy-data-system could become a service its-self (some day, in the long term).
+And most server-side code from cozy-home too. Maybe, we can even unify all of
+our nodejs code to run in the same processus (I'm dreaming).
+
+### 3. As express/americano middlewares
+
+It's the simple way I see to make this happen.
```

Comment:
I would say as an express app. You can mount apps inside another app with Express.